integration/test/performance_tracker/transmitter/transmitter.go

```diff
@@ -0,0 +1,237 @@
+	// fmt.Printf("Metric: %+v \n OG:%+v \n",metrics, packet.Metrics)
+	item, err := dynamodbattribute.MarshalMap(packet)
+	if err != nil {
+		panic(err)
+	}
+	DBitem := &dynamodb.PutItemInput{
+		Item:      item,
+		TableName: aws.String(transmitter.DataBaseName),
+	}
+	// fmt.Println(DBitem)
+	// return err
+	_, err = transmitter.DynamoDbClient.PutItem(DBitem)
+	if err != nil {
+		fmt.Printf("Couldn't add item to table. Here's why: %v\n", err)
+	}
+	return fmt.Sprintf("%v",item),err
+}
+/*
+TableExist()
+Desc: Checks if the the table exist and returns the value
+//https://github.com/awsdocs/aws-doc-sdk-examples/blob/05a89da8c2f2e40781429a7c34cf2f2b9ae35f89/gov2/dynamodb/actions/table_basics.go
+*/
+func (transmitter * TransmitterAPI) TableExist() (bool,error){
+	l,err := transmitter.ListTables()
+	for i:=0; i< len(l); i++{
+		if transmitter.DataBaseName == l[i]{
+			return true,nil
+		}
+	}
+	return false,err
+
+}
+/*
+RemoveTable()
+Desc: Removes the table that was craeted with initialization. 
+Side effects: Removes a Dynamodb table
+*/
+func (transmitter * TransmitterAPI) RemoveTable() error{
+	input := dynamodb.DeleteTableInput{TableName: aws.String(transmitter.DataBaseName)}
+
+	_,err:=transmitter.DynamoDbClient.DeleteTable(&input)
+	if err !=nil{
+		fmt.Println(err)
+	}
+	return err
+}
+// ListTables lists the DynamoDB table names for the current account.
+func (transmitter * TransmitterAPI) ListTables() ([]string, error) {
+	var tableNames []string
+	input := &dynamodb.ListTablesInput{}
+	tables, err := transmitter.DynamoDbClient.ListTables(
+		input)
+	if err != nil {
+		fmt.Printf("Couldn't list tables. Here's why: %v\n", err)
+	} else {
+		for  i:=0; i< len(tables.TableNames); i++{
+			tableNames =  append(tableNames,*tables.TableNames[i])
+		}
+		
+	}
+	return tableNames, err
+}
+
+func (transmitter * TransmitterAPI) SendItem(data []byte) (string,error) {
+	// return nil
+	packet, err := transmitter.Parser(data)
+	if err != nil{
+		return "",err
+	}
+	sentItem,err := transmitter.AddItem(packet) 
+	return sentItem,err
+}
+
+func (transmitter * TransmitterAPI) Parser(data []byte) (map[string]interface{},error){
+	dataHolder := collectorData{}
+	err := json.Unmarshal(data,&dataHolder)
+	if err !=nil{
+		return nil,err
+	}
+	packet := make(map[string]interface{})
+	//temp solution
+	packet["Hash"] = fmt.Sprintf("%d",time.Now().UnixNano())
+	/// will remove
+	for _,rawMetricData:= range dataHolder{
+		numDataPoints := float64(len(rawMetricData.Timestamps))
+		sum :=0.0
+		for _,val := range rawMetricData.Values {
+			sum += val
+		}
+		avg :=  sum /numDataPoints
+		// calculate diff between mean and values 
+		diffSum := 0.0
+		for _,val := range rawMetricData.Values{
+			diffSum = diffSum +  (avg - val)
+		}
+		
+		metric := Metric{
+			Average: avg,
+			StandardDev: math.Sqrt(math.Pow(diffSum,2)/float64(numDataPoints)),
+			Period: int(TIME_CONSTANT/(numDataPoints)),
+			Data:  rawMetricData.Values}
```

Comment:
I don't think we need to do any of this math ourselves. Take a look at GetMetricStatistics https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/US_SingleMetricPerInstance.html We also discussed in the last review that we think that the p99 stat is more valuable than the mean.

Reply:
If we move to use GetMetricStatistics, I don't think most of this is necessary.